lib/avo/fields/location_field.rb

```diff
@@ -0,0 +1,70 @@
+# frozen_string_literal: true
+
+module Avo
+  module Fields
+    class LocationField < BaseField
+      attr_reader :stored_as
+
+      def initialize(id, **args, &block)
+        hide_on :index
+
+        super(id, **args, &block)
+
+        @stored_as = args[:stored_as].present? ? args[:stored_as] : nil # You can pass it an array of db columns [:latitude, :longitude]
+      end
+
+      def as_lat_long?
+        stored_as.is_a?(Array) && stored_as.count == 2
+      end
+
+      def as_lat_long_id(get: nil)
+        if get == :lat
+          "#{id}[#{stored_as.first}]"
+        elsif get == :long
+          "#{id}[#{stored_as.last}]"
+        end
+      end
+
+      def as_lat_long_placeholder(get: nil)
+        if get == :lat
+          placeholder = "Enter #{stored_as.first}"
+        elsif get == :long
+          placeholder = "Enter #{stored_as.last}"
```

Comment:
You can skip assigning these to a variable, since they will be the return value of the `if` statement / the method itself.

```suggestion
          "Enter #{stored_as.last}"
```